### ISSUE TITLE
further understading for 2d dataset plan_and_preprocess

### ISSUE THREAD

**Issue by SanBast** [MIC-DKFZ/nnUNet#1840]
Hello,

I was going to train nnUNet with a custom dataset that is quite similar to the MSD Task05_Prostate. The only difference is that I'm dealing with 2d slices instead of the entire temporal volumetric data. An example:

imagesTr:
- prostate_00_slice_009.nii.gz
- prostate_00_slice_010.nii.gz
- and so on

I followed the tutorial [here](https://github.com/MIC-DKFZ/nnUNet/blob/nnunetv1/nnunet/dataset_conversion/Task120_Massachusetts_RoadSegm.py). Of course I know I'm working with niftii files already, (instead of the suggested png) and I created a function accordingly, based on the one used in the tutorial to generate niftii files. Moreover, I checked that the shape and the file locations were both right, and everything run smoothly.

However, when I decide to run `nnUNet_plan_and_preprocess -t 5 -pl3d None` it outputs the following error:

```
Traceback (most recent call last):

  File "/usr/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
    
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 51, in starmapstar
    return list(itertools.starmap(args[0], args[1]))
    
  File "/data/marciano/my_envs/FORE/lib/python3.8/site-packages/nnunet/preprocessing/cropping.py", line 174, in load_crop_save
    raise e

  File "/data/marciano/my_envs/FORE/lib/python3.8/site-packages/nnunet/preprocessing/cropping.py", line 165, in load_crop_save
    data, seg, properties = self.crop_from_list_of_files(case[:-1], case[-1])

  File "/data/marciano/my_envs/FORE/lib/python3.8/site-packages/nnunet/preprocessing/cropping.py", line 155, in crop_from_list_of_files
    data, seg, properties = load_case_from_list_of_files(data_files, seg_file)

  File "/data/marciano/my_envs/FORE/lib/python3.8/site-packages/nnunet/preprocessing/cropping.py", line 65, in load_case_from_list_of_files
    data_itk = [sitk.ReadImage(f) for f in data_files]

  File "/data/marciano/my_envs/FORE/lib/python3.8/site-packages/nnunet/preprocessing/cropping.py", line 65, in <listcomp>
    data_itk = [sitk.ReadImage(f) for f in data_files]

  File "/data/marciano/my_envs/FORE/lib/python3.8/site-packages/SimpleITK/extra.py", line 355, in ReadImage
    return reader.Execute()

  File "/data/marciano/my_envs/FORE/lib/python3.8/site-packages/SimpleITK/SimpleITK.py", line 8438, in Execute
    return _SimpleITK.ImageFileReader_Execute(self)

RuntimeError: Exception thrown in SimpleITK ImageFileReader_Execute: /tmp/SimpleITK/Code/IO/src/sitkImageReaderBase.cxx:97:
sitk::ERROR: The file "/data/marciano/nnUnet/nnUNet_raw_data_base/nnUNet_raw_data/Task005_Prostate/imagesTr/prostate_01_slice_016_0002.nii.gz" does not exist.
```

In summary, during the run_cropping function for some reasons the cases found by the script are 3 (0000,0001,0002) instead of 2 (0000,0001). I assume that the cases refer to the _channels_ of each image, that are 2 (e.g. before processing for an image X I have the shape 2,320,320) as you can also find in the prostate MSD dataset.

I tried to debug more but I still got stucked there.

Can you please guide me through this?

Thanks you very much for your patience in reading this.
Sincerely,
V

**Comment by SanBast** [comment]
I'd also like to add that in my nnunet-ready dataset folder there's no filename showing 0002, but 0000 and 0001 only.

**Comment by SanBast** [comment]
Bug found, my fault. It was the way for which we need to input the 2d niftii data, and a wrong indexing of the dictionary keys.
Thanks for your attention, and sorry.

V
